File: app/src/main/java/shts/jp/android/nogifeed/fragments/CalenderFragment.java
package shts.jp.android.nogifeed.fragments;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import shts.jp.android.nogifeed.R;

public class CalenderFragment extends Fragment {

    private static final String TAG = CalenderFragment.class.getSimpleName();

    public static CalenderFragment newInstance() {
        return new CalenderFragment();
    }

    private WebView webView;

    @SuppressLint("SetJavaScriptEnabled")
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_calender, null);
        webView = (WebView) view.findViewById(R.id.browser);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                if (Uri.parse(url).getHost().equals("sp.nogizaka46.com")) {
                    return false;
                }
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
                startActivity(intent);
                return true;
            }
        });
        webView.loadUrl("http://sp.nogizaka46.com/q?i=calendar/index&ymd");
        return view;
    }

    public boolean goBack() {
        if (webView.canGoBack()) {
            webView.goBack();
            return true;
        }
        return false;
    }
}

File: app/src/main/java/shts/jp/android/nogifeed/fragments/CalenderFragment.kt
package shts.jp.android.nogifeed.fragments

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import kotlinx.android.synthetic.main.fragment_calender.*
import shts.jp.android.nogifeed.R

class CalenderFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater?.inflate(R.layout.fragment_calender, null)
    }

    @SuppressLint("SetJavaScriptEnabled")
    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        webView.settings.javaScriptEnabled = true
        webView.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
                request?.let {
                    if (it.url?.host == "sp.nogizaka46.com") {
                        return false
                    }
                    startActivity(Intent(Intent.ACTION_VIEW, it.url))
                    return true
                }
                return super.shouldOverrideUrlLoading(view, request)
            }
        }
        webView.loadUrl(URL)
    }

    fun goBack(): Boolean {
        if (webView.canGoBack()) {
            webView.goBack()
            return true
        }
        return false
    }

    companion object {
        private val URL: String = "http://sp.nogizaka46.com/qschedule?type=monthly&ymd=&cat=&nm="

        fun newInstance(): CalenderFragment {
            return CalenderFragment()
        }
    }
}
